tools/run_functional_tests.py

```diff
@@ -0,0 +1,340 @@
+
+
+def wait_for_response():
+    data = ''
+    no_response_counter = 0
+
+    while (no_response_counter <= MAX_GET_LINE_RETRIES):
+        sleep(.005)
+        data = read_output_serial()
+        if (data):
+            return data
+        no_response_counter += 1
+
+    return None
+
+
+#
+# MARK: - Functions
+#
+
+TESTS_FUNCTIONAL_BIN_FILES = list()
+
+
+def list_bin_files():
+    set = list()
+    for root, dirs, files in os.walk(TESTS_FUNCTIONAL_ROOT_DIRECTORY):
+        for filename in files:
+            if filename.endswith(TESTS_BIN_EXTENSION):
+                set.append(
+                    os.path.join(root, filename))
+
+    return set
+
+
+TESTS_FUNCTIONAL_BIN_FILES = list_bin_files() if args.all else args.bin_files
+
+FLASH_ERASE_FLAG = args.flash_erase
+
+
+def warningprint(*args, **kwargs):
+    print(Fore.YELLOW + "\n⚠️ Warning : " +
+          " ".join(map(str, args))+Style.RESET_ALL, **kwargs)
+
+
+#
+# MARK: - Class Test
+#
+
+class Test:
+
+    def __init__(self, path):
+        self.path = path
+
+    def generate_result_file(self):
+        def define_path(source_path):
+            base, ext = os.path.splitext(source_path)
+            timestamp = time.time()
+            date = str(datetime.datetime.fromtimestamp(
+                timestamp))
+            date = date.replace(':', '.')
+            date = date.replace(' ', '_')
+            new_extension = ".txt"
+            target_path = base + "_" + date + new_extension
+            return target_path
+
+        def create_file(path):
+            try:
+                file = open(path, "w")
+            except OSError as e:
+                print("Could not create or open file: " + path)
+                print("Error: " + e)
+                sys.exit(1)
+            file.close()
+
+        self.result_filepath = define_path(self.path)
+        create_file(self.result_filepath)
+
+    def edit_result_file(self, data):
+        result_filepath = self.result_filepath
+        try:
+            with open(result_filepath, "a") as file:
+                file.write(data.strip() + '\n')
+        except FileNotFoundError as e:
+            print("The file: " + result_filepath + "doesn\'t exist")
+            print("Error: " + e)
+            sys.exit(1)
+
+    def print_result_file(self):
+        result_filepath = self.result_filepath
+        try:
+            with open(result_filepath, "r") as file:
+                data = file.read()
+            if (data):
+                print(data)
+            else:
+                warningprint("No data !")
+        except FileNotFoundError as e:
+            print("The file: " + result_filepath + "doesn\'t exist")
+            print("Error: " + e)
+            sys.exit(1)
+
+    def flash(self):
+        print("\n")
+        print(self.path)
+        CMD_FLASH = "openocd -f interface/stlink.cfg -c 'transport select hla_swd' -f target/stm32f7x.cfg -c 'program " + \
+            self.path+" 0x08000000' -c exit"
```

Comment:
This script has a lot of operation and it's hard to review as it is.

Just a comment, some information and command as `CMD_FLASH` should have same source.